Source/Particles/Collision/BinaryCollision/NuclearFusion/SingleNuclearFusionEvent.H

```diff
@@ -118,14 +118,15 @@ void SingleNuclearFusionEvent (const amrex::ParticleReal& u1x, const amrex::Part
 
     // Square of the norm of the momentum of one of the particles in the center of mass frame
     // Formula obtained by inverting E^2 = p^2*c^2 + m^2*c^4 in the COM frame for each particle
-    auto constexpr pow3 = [](double const x) { return x*x*x; };
-    const amrex::ParticleReal p_star_sq =
-            E_star_sq*inv_four_pr*inv_csq - (m1_sq + m2_sq)*c_sq*inv_two_pr +
-            pow3(c_sq) * inv_four_pr * pow2(m1_sq - m2_sq) / E_star_sq;
+    // The expression below is specifically written in a form that avoids returning
+    // small negative numbers due to machine precision errors, for low-energy particles
+    const amrex::ParticleReal E_ratio = E_star/((m1 + m2)*c_sq);
```

Comment:
Do we have a complete guarantee that this number will never be strictly smaller than 1? E.g. if all the momenta are 0, this corresponds to the ratio between sqrt(E_rest**2) and E_rest so it's not obvious which of the two numbers will be smaller within machine precision.

Reply:
That definitely makes sense to me.
I think I'm fine with both leaving it as it is (and reconsider if we encounter issues in the future) and with anticipating possible future issues by adding a line like `E_star = max( E_star, (m1 + m2)*c_sq )`.

Feel free to do what you think is best, I'll approve the PR in the meantime.